src/Shared/ExceptionHandling.cs

```diff
@@ -34,12 +34,7 @@ namespace Microsoft.Build.Shared
     /// </summary>
     internal static class ExceptionHandling
     {
-        private static readonly string s_debugDumpPath;
-
-        static ExceptionHandling()
-        {
-            s_debugDumpPath = GetDebugDumpPath();
-        }
+        private static readonly string s_debugDumpPath = GetDebugDumpPath();
```

Comment:
This change may be counter-productive. Presumably the debug dump path is needed only if there actually was an exception. With this change we may be initializing it eagerly which is wasted work when no exception is thrown. We can afford to pay the small cost of precise (i.e. non-`beforefieldinit`) initialization on error paths.

Can you please perform a simple test to see if `GetDebugDumpPath` runs without your changes when the build succeeds?

Reply:
I feel like you teach me something new about C# every week 😁

When would static field be initialized without the static constructor being called? ["If static field variable initializers are present in the class of the static constructor, they're executed in the textual order in which they appear in the class declaration. The initializers run immediately prior to the execution of the static constructor."](https://docs.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/static-constructors) made it sound like they always happen together.

Reply:
They do happen together. Field initializers become part of the static ctor at IL level. It's just that depending on whether the class does or doesn't have a static ctor declared in C#, the code may run a different time. Let me link an older article explaining the difference: https://csharpindepth.com/articles/BeforeFieldInit

Reply:
@ladipro I tested with a small .net framework project. It didn't hit `GetDebugDumpPath` when when the build succeeds before or after the change.

Reply:
Thank you! Let's go with the change then. There's still a possibility that we will initialize more eagerly - depending on the runtime and the exact code path - plus the initialization has a side-effect of creating a directory (see the `DebugUtils` static ctor) but it's under a debug switch so no biggie.